megamek/src/megamek/common/GunEmplacement.java

```diff
@@ -612,18 +594,18 @@ public void setDeployed(boolean deployed) {
      */
     @Override
     public double getArmorRemainingPercent() {
-        if(getPosition() == null) {
+        if (getPosition() == null) {
             return 1.0;
         }
         
         Building occupiedStructure = getGame().getBoard().getBuildingAt(getPosition());
         
         // we'll consider undeployed emplacements to be fully intact
-        if((occupiedStructure == null) || (initialBuildingCF + initialBuildingArmor == 0)) {
+        if ((occupiedStructure == null) || (initialBuildingCF + initialBuildingArmor == 0)) {
             return 1.0;
         }
         
-        return (occupiedStructure.getCurrentCF(getPosition()) + occupiedStructure.getArmor(getPosition())) / 
-                (initialBuildingCF + initialBuildingArmor);
+        return (1.0 * (occupiedStructure.getCurrentCF(getPosition()) + occupiedStructure.getArmor(getPosition())))
```

Comment:
Is this a bug fix?

Reply:
It is, it was trying to return a double from an int/int.

Reply:
More obvious to cast the denominator to double. I presume this means that gun emplacements never showed the correct amount of armor remaining? (always 0)

Reply:
Swapped, and I believe so. This was a catch from my IDE.